odoo/model/odoo.go

```diff
@@ -0,0 +1,16 @@
+package model
+
+import "github.com/vshn/appuio-odoo-adapter/odoo"
+
+type Odoo struct {
+	client  *odoo.Client
+	session *odoo.Session
+}
+
+// NewOdoo creates a new Odoo client.
+func NewOdoo(session *odoo.Session, client *odoo.Client) *Odoo {
+	return &Odoo{
+		client:  client,
+		session: session,
+	}
+}
```

Comment:
This Odoo type (naming up for debate) is basically the same as a client, but with the session stored as a field. This should improve developer UX by not having to pass in the session instance to every call.

Alternative solution would be that `odoo.Client` stores the session as a field in the `odoo.Client` after calling `Login()`. This means that methods like `CreateGenericModel(..., session, ...)` can have the session parameter removed, but one needs  to know to call `Login()` before.

Thoughts?

Reply:
I like the split between more generic raw client and the developer friendly one. I've seen it often e.g. prometheus api client.

Reply:
Without your comment I was very confused about this struct. I don't find this very intuitive, but I see the issue of having to pass it to every call. I would personally prefer to store the session as part of the client. Two other alternatives to your approach would be to

1. Define the current methods of `Odoo` on the `Session` struct.
2. Store the session in the context. This is IMO one of the few situations were this is acceptable

All approaches have pros and cons. I would also be fine with keeping it as is.

Reply:
Thanks for you suggestions. I think I'll go with `Client -> Login() creates Session -> Session has generic methods -> Odoo type takes Session`